src/components/forms/riddleUpsertForm/RiddleShareForm.tsx

```diff
@@ -0,0 +1,96 @@
+import { Box, Button, Stack, Typography } from '@mui/material';
+import type { UseFormReturn } from 'react-hook-form-mui';
+import { FormContainer, TextFieldElement } from 'react-hook-form-mui';
+import React from 'react';
+
+import type { RiddleUpsertDetail } from '../../../utils/Types';
+import { RadioButtonFormComponentBroad } from '../common/RadioButtonFormComponentBroad';
+import { CopyContentButton } from '../common/CopyContentButton';
+import { AutocompleteUsers } from '../common/AutocompleteUsers';
+
+type Props = {
+	formContext: UseFormReturn<RiddleUpsertDetail>;
+	riddleName: string | null;
+	onSubmitFinal: (data: RiddleUpsertDetail) => void;
+	handleBack: () => void;
+	onCancel: () => void;
+};
+
+export const RiddleShareForm = ({
+	formContext,
+	riddleName,
+	onSubmitFinal,
+	handleBack,
+	onCancel
+}: Props) => {
+	const { watch } = formContext;
+	const watchIsPublic = watch('sharingInformation.visibility');
+	const watchLink = watch('linkId');
+
+	return (
+		<FormContainer onSuccess={onSubmitFinal} formContext={formContext}>
+			<Stack gap={2} sx={{ minWidth: { md: 500 } }}>
+				<Typography variant="h2">{riddleName}</Typography>
+				<Typography variant="subtitle1">
+					The last thing you need to do to finish your riddle is to publish it.
```

Comment:
This was changed against Figma because it would be more complicated to store and then make additional updates.